### ISSUE TITLE
[M68k] Add floating point support

### ISSUE THREAD

**Issue by mshockwave** [llvm/llvm-project#61744]
Add supports for floating pointer registers, basic instructions and inline assembly.

**Comment by 0x59616e** [comment]
Hello @mshockwave ,

May I inquire if we have any plans or milestones established? Additionally, is there a corresponding to-do list for said milestones?

**Comment by mshockwave** [comment]
> Hello @mshockwave ,
> 
> May I inquire if we have any plans or milestones established? Additionally, is there a corresponding to-do list for said milestones?

Yeah I will put more detailed milestones for this. In the short term, we have to pass llvm-test-suite, which relies on inline assembly with floating point constructions. For instance:
```c
__asm__ volatile ("fmove%.l %0, %!" : : "dm" (cw))
```
So for that, we at least need floating pointer register definitions and MC supports for commonly-used floating point instructions.

**Comment by mshockwave** [comment]
UPDATE: just updated the description to add (preliminary) roadmap for this task.

**Comment by 0x59616e** [comment]
It appears that this is a large-scale project. I'm wondering if it's possible to divide these tasks and list them in a table format ? This could help us keep track of which items are currently under development and avoid any potential duplication of effort.

Something like this:
| item | picker|
|-----|-------|
| register support| aaa |
| instruction mc support | bbb |
| inline support                 | ccc |

**Comment by mshockwave** [comment]
> It appears that this is a large-scale project. I'm wondering if it's possible to divide these tasks and list them in a table format ? This could help us keep track of which items are currently under development and avoid any potential duplication of effort.
> 
> Something like this:
> 
> item	picker
> register support	aaa
> instruction mc support	bbb
> inline support	ccc

Yes I agree. I've created issues for these tasks. Please put yourself as assignee if you want to work on that.

**Comment by 0x59616e** [comment]
Thanks ! Looks like we have so much fun to do.

**Comment by glaubitz** [comment]
Any news on the M68k backend?
